lib/onigumo.ex

```diff
@@ -9,11 +9,19 @@ defmodule Onigumo do
     HTTPoison.start()
     http = http_client()
 
-    load_urls(@input_filename)
-    |> Enum.map(&download(http, &1))
+    download(http)
   end
 
-  def download(http_client, url) do
+  def download(http_client) do
+    urls = load_urls(@input_filename)
+    download(http_client, urls)
+  end
+
+  def download(http_client, urls) when is_list(urls) do
+    Enum.map(urls, &download(http_client, &1))
+  end
+
+  def download(http_client, url) when is_bitstring(url) do
```

Comment:
I am not sure about using `is_bitstring` as guard to strings. Please, consider this article https://stackoverflow.com/questions/54161232/use-is-bitstring-or-is-binary-in-elixir-guard-for-string.
If I've read carefully and I've understood well the guard `is_binary` is more suitable for strings.
```suggestion
  def download(http_client, url) when is_binary(url) do
```

Reply:
Great point! I didn’t know about the difference and that `is_binary` is a convention. Thanks!

Reply:
Fixed. https://github.com/Glutexo/onigumo/pull/29/commits/684591a2cb3957e2424f86a145f10a07933134ab